code/modules/mob/living/silicon/robot/robot_items.dm

```diff
@@ -1,3 +1,533 @@
+				desc = "An advanced integrated scanning device designed for medical use. Can quickly switch between Health, Reagent and Antibody Analyzer modes. Currently set to Health Analyzer mode."
+				mode = 1
+			if(1)//Reagent Analyzer Mode
+				name = "Med Scanner (Reagent Analyzer Mode)"
+				icon_state = "adv_spectrometer"
+				user << "\blue Med Scanner set to Reagent Analyzer mode."
+				desc = "An advanced integrated scanning device designed for medical use. Can quickly switch between Health, Reagent and Antibody Analyzer modes.  Currently set to Reagent Analyzer mode"
+				mode = 2
+			if(2)//Antibody Analyzer Mode
+				name = "Med Scanner (Antibody Analyzer Mode)"
+				icon_state = "antibody"
+				user << "\blue Med Scanner set to Antibody Analyzer mode."
+				desc = "An advanced integrated scanning device designed for medical use. Can quickly switch between Health, Reagent and Antibody Analyzer modes. Currently set to Antibody Analyzer mode."
+				mode = 3
+		playsound(src.loc, 'sound/effects/pop.ogg', 50, 0)
+
+	attack(mob/target as mob, mob/user as mob)
+		var obj/item/device/S = scanners[mode]
+		S.attack(target, user)
+
+	afterattack(mob/target as mob, mob/user as mob)
+		var obj/item/device/S = scanners[mode]
+		S.afterattack(target, user)
+
+
+/obj/item/weapon/cyborg/bodybag_containment_unit
+	name = "Bodybag Containment Unit"
+	desc = "An integrated bodybag containment and deployment unit. Can store up to ten bodybags and cryobags in total."
+	icon = 'icons/obj/bodybag.dmi'
+	icon_state = "bodybag_closed"
+	var/bags_max = 10
+	var/bodybags_stored = 10
+	var/cryobags_stored = 0
+	var/bags_stored = 10
+	var/mode = "bodybag"
+
+/obj/item/weapon/cyborg/bodybag_containment_unit/attack_self(mob/living/user as mob)
+	if(!bags_stored)
+		playsound(user, 'sound/weapons/emptyclick.ogg', 50, 0)
+		user << "<span class='notice'>You have no bags stored.</span>"
+
+	if(mode == "stasis bag")
+		if(!cryobags_stored)
+			playsound(user, 'sound/weapons/emptyclick.ogg', 50, 0)
+			user << "<span class='notice'>You have no [mode]s stored.</span>"
+			return
+		new /obj/structure/closet/body_bag/cryobag(user.loc)
+		cryobags_stored -= 1
+
+	else if(mode == "bodybag")
+		if(!bodybags_stored)
+			playsound(user, 'sound/weapons/emptyclick.ogg', 50, 0)
+			user << "<span class='notice'>You have no [mode]s stored.</span>"
+			return
+		new /obj/structure/closet/body_bag(user.loc)
+		bodybags_stored -= 1
+
+	bags_stored = bodybags_stored + cryobags_stored
+	playsound(src, 'sound/items/Deconstruct.ogg', 50, 1)
+	user << "<span class='notice'>You deploy a [mode]. You currently have [bodybags_stored] bodybag(s) stored and [cryobags_stored] stasisbag(s) stored.</span>"
+	return
+
+
+/obj/item/weapon/cyborg/bodybag_containment_unit/verb/deploy_mode()
+
+	set name = "Set Bodybag Deployment Mode"
+	set desc = "Allows you to set the Bodybag Containment Unit to deploy cryobags or bodybags."
+	set category = "Robot Commands"
```

Comment:
I would be suprised if this even works. It should have a 'set loc in owner' somewhere in here.